web/src/components/role/AddUpdateRole.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+  <q-card class="my-card">
+    <q-card-section>
+      <div v-if="beingUpdated" class="text-h6">Update role</div>
+      <div v-else class="text-h6">Add role</div>
+    </q-card-section>
+    <q-card-section>
+      <q-form ref="addRoleForm" @submit="onSubmit">
+        <q-input
+          v-model="roleData._id"
+          dense
+          borderless
+          filled
+          :readonly="beingUpdated"
+          :disabled="beingUpdated"
+          :bg-color="disableColor"
+          :label="t('role.id')"
+          :rules="[validateRoleID]"
+        />
+        <q-input
+          v-model="roleData.name"
+          dense
+          borderless
+          filled
+          :label="t('role.name')"
+          :rules="[validateRoleName]"
+        />
+        <q-select
+          class="q-field--with-bottom"
+          v-model="roleData.permission"
+          :options="permissions"
+          dense
+          multiple
+          use-chips
+          borderless
+          filled
+          :label="t('role.permission')"
+        />
+
+        <q-btn
+          no-caps
+          class="q-mb-md"
+          color="primary"
+          type="submit"
+          icon="add"
+          label="Save Role"
+        />
+      </q-form>
+    </q-card-section>
+  </q-card>
+</template>
+
+<script>
+import { defineComponent, ref } from "vue";
+import roleService from "../../services/role";
+import permissionService from "../../services/permission";
+import { useI18n } from "vue-i18n";
+
+const defaultValue = () => {
+  return {
+    _id: "",
+    name: "",
+    role: "",
+    permission: [],
+  };
+};
+
+export default defineComponent({
+  name: "ComponentAddUpdateRole",
```

Comment:
Maybe we can leave out the Component prefix ?

Reply:
[copy](https://github.com/zinclabs/zinc/blob/main/web/src/components/user/AddUpdateUser.vue#L101)